packages/nextjs/components/scaffold-eth/Address.tsx

```diff
@@ -76,6 +78,10 @@ export const Address = ({ address, disableAddressLink, format }: TAddressProps)
       </div>
       {disableAddressLink ? (
         <span className="ml-1.5 text-lg font-normal">{displayAddress}</span>
+      ) : getTargetNetwork().id === chains.hardhat.id ? (
+        <span className="ml-1.5 text-lg font-normal">
+          <Link href={`/blockexplorer/address/${address}`}>{displayAddress}</Link>
```

Comment:
We can use `blockExplorerAddressLink` directly since we have already added the logic to handle `hardhat` in `getBlockExplorerAddressLink` 

NITPICK : Also instead of `chains.hardhat.id` we can : 
```ts
import {hardhat} from "wagmi/chains"
hardhat.id
```
which might help in tree shaking